### DISCUSSION TITLE
Issue with IBM Wazi deploy intallation in Mainframe

### DISCUSSION THREAD

**Discussion by ratnamGT** [IBM/dbb#264] [Q&A]
I am trying to set up and configure IBM Wazi deploy in Mainframe  . I configured and created the python envs,ZOAU and wazi environment.I am getting below installation error when I am trying to install and configure IBM Wazi deploy with the following command after tar file extraction .We are not able to run any wazi deploy commands due to this error.
pip3 install wazideploy-3.0.0-py3.11-none-any.whl
![image](https://github.com/user-attachments/assets/21d2ca8d-511c-4e4a-8fab-f4d8198e5524)
![image](https://github.com/user-attachments/assets/2993e1b5-9485-4bf8-9701-4f8745a89527)

We are using Python 3.11.2, ZOAU version 1.2.3,IBM Wazi deploy version-3.0.0 .Could you please help us to resolve the issue?

**Comment by dennis-behm** [comment]
Hi @ratnamGT - Please note that this repository is not the official IBM support channel. You might want to open a support case for this.

Please be aware of the available fixes for Wazi Deploy as well - https://www.ibm.com/support/pages/node/7056432

**Comment by dennis-behm** [comment]
@ratnamGT please apply the latest PTF for Wazi Deploy. This should resolve the problem.

**Comment by ratnamGT** [comment]
@dennis-behm  Thanks for response .We are using IBM Wazi deploy version 3.0.0 and these PTFs are for version 3.0.1 and 3.0.2 . This issue is coming while we setup the Wazi deploy after ZOAU and python configuration steps when we execute "pip3 install wazideploy-3.0.0-py3.11-none-any.whl"

**Comment by jbyibm** [comment]
Due to the latest version of jsonschema-4.18.2 that is not pure Python code, you need to install jsonschema before wazideploy with:
```
pip3 install jsonschema==4.17.3
```

This workaround is only for version 3.0.0. Please update to the latest version after.

**Comment by ratnamGT** [comment]
@jbyibm  Thanks for your response. The command is working and Wazi deploy installed but I am not able to see any output when I run the command Wazideploy- to verify the availability of commands .it is throwing error that "wazideploy-: FSUM7351 not found".
Could you please help us?

**Reply by dennis-behm** [reply]
I assume you need to add the `bin` dir to your path, that they can be resolved automatically.

```
DBEHM:/u/dbehm: >type wazideploy-deploy
wazideploy-deploy is /u/dbehm/.local/bin/wazideploy-deploy
```

in my .profile
```
export PATH=/u/dbehm/.local/bin:$PATH
```

**Reply by ratnamGT** [reply]
Hi @dennis-behm , I can see that bin directory has 3 commands only wazideploy-deploy,wazideploy-generate,wazideploy-package .I can execute wazideploy-deploy command and also remaining 3 but wazideploy-evidence and wazideploy- command is not working .May be because it is not available in bin directory .Could you please lte us know why these commands are missing ? Or what to add here in u/ibmuser/.profile file ?
![image](https://github.com/user-attachments/assets/1e7d8459-6612-43ae-bd87-77f081abe8a5)
![image](https://github.com/user-attachments/assets/09922679-b6d0-44b9-ab60-ca0affd23683)

**Reply by dennis-behm** [reply]
@ratnamGT - We are now back to the problem that you are on 3.0.0. This version does not have the wazideploy-evidence command.

**Reply by ratnamGT** [reply]
@dennis-behm  Thanks for clarification , Does 3.0.0 not have wazi-deploy- command also to verify the availability of command . Will it effect any deployment if we use 3.0.0 version?
Should I consider that 3.0.0 is successfully installed or any issue there?

**Comment by ratnamGT** [comment]
Hi @jbyibm ,

Could you please share the EOLE53.yml file and Wazi deploy 3.2 documentation ? I could not find the documentation for wazideploy-deploy command in IBM documentation and also the git repository git@github.com:jbyibm/cics-genapp.git ?

**Reply by dennis-behm** [reply]
Hi @ratnamGT ,

please see the documentation link for Wazi Deploy including all the commands at:
https://www.ibm.com/docs/en/developer-for-zos/16.0?topic=deploy-whats-new-in-wazi

JYB has published his environment files as samples at
https://github.com/jbyibm/cics-genapp/tree/main/wazideploy-samples/external-repos

**Comment by ratnamGT** [comment]
Hi @dennis-behm  Thanks for response. This link does not have explanation for wazideploy-deploy command and argument details like other commands . As I told that I do not have wazideploy-evidence command in 3.0.0 version and the evidences file is the parameter in the wazideploy-deploy command ,I just want to check weather it is mandatory parameter or I can run the deployment without it.
https://www.ibm.com/docs/en/developer-for-zos/16.0?topic=commands-wazi-deploy-generation-command
Also I could not find the documentation that how to configure the environment.yml file to deploy the package as per my machine configuration.
Could you please help me?

**Reply by dennis-behm** [reply]
@ratnamGT,

well it does have the necessary explanations. The deploy command is depending on the selected translator/technlogoy - either Ansible or Python. You find it in this section: https://www.ibm.com/docs/en/developer-for-zos/16.0?topic=reference-wazi-deploy-ansible-python-translators

The Evidence command is important to track deployment activities. Using the wazideploy-deploy command just deploys the package. So at minimum you need `wazideploy-generate` and `wazideploy-deploy`.

Please find [sample environment files](https://github.com/jbyibm/cics-genapp/blob/main/wazideploy-samples/external-repos/environment-conf/python/environment-zos.yml) in JYBs repository. Generally, it's just a file that contains the evironment `spec`
https://www.ibm.com/docs/en/developer-for-zos/16.0?topic=translators-generic-environment-variables-ansible-python-building-blocks#wd_templ_env_files__python_var__title__1

**Comment by ratnamGT** [comment]
Thanks @dennis-behm for the documentation. I will try to implement it
